embedded-graphics/src/primitives/sector.rs

```diff
@@ -0,0 +1,798 @@
+        let line_b = LinearEquation::from_point_angle(self.center(), self.angle_end());
+        let draw_above_a = (angle_start < ANGLE_90DEG) ^ negative_sweep;
+        let draw_above_b = (angle_end >= ANGLE_90DEG) ^ negative_sweep;
+        let sweep = self.angle_sweep.abs();
+
+        let side_a = line_a.side(&point);
+        let side_b = line_b.side(&point);
+
+        let correct_a_side = draw_above_a ^ (side_a == LineSide::BELLOW);
+        let correct_b_side = draw_above_b ^ (side_b == LineSide::BELLOW);
+
+        if sweep < ANGLE_180DEG {
+            correct_a_side && correct_b_side
+        } else if sweep < ANGLE_360DEG {
+            correct_a_side || correct_b_side
+        } else {
+            true
+        }
+    }
+}
+
+impl Dimensions for Sector {
+    fn bounding_box(&self) -> Rectangle {
+        Rectangle::new(self.top_left, Size::new(self.diameter, self.diameter))
+    }
+}
+
+impl Transform for Sector {
+    /// Translate the sector from its current position to a new position by (x, y) pixels,
+    /// returning a new `Sector`. For a mutating transform, see `translate_mut`.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Sector;
+    /// # use embedded_graphics::prelude::*;
+    /// let sector = Sector::new(Point::new(5, 10), 10, 0.0.deg(), 90.0.deg());
+    /// let moved = sector.translate(Point::new(10, 10));
+    ///
+    /// assert_eq!(moved.top_left, Point::new(15, 20));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            top_left: self.top_left + by,
+            ..*self
+        }
+    }
+
+    /// Translate the sector from its current position to a new position by (x, y) pixels.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Sector;
+    /// # use embedded_graphics::prelude::*;
+    /// let mut sector = Sector::new(Point::new(5, 10), 10, 0.0.deg(), 90.0.deg());
+    /// sector.translate_mut(Point::new(10, 10));
+    ///
+    /// assert_eq!(sector.top_left, Point::new(15, 20));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.top_left += by;
+
+        self
+    }
+}
+
+/// Iterator over all points inside the sector.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct Points {
+    iter: DistanceIterator,
+    threshold: u32,
+}
+
+impl Points {
+    fn new(sector: &Sector) -> Self {
+        let threshold = circle::diameter_to_threshold(sector.diameter);
+
+        Self {
+            iter: DistanceIterator::new(&sector),
+            threshold,
+        }
+    }
+}
+
+impl Iterator for Points {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let threshold = self.threshold;
+        self.iter
+            .find(|(_, distance)| *distance < threshold)
+            .map(|(point, _)| point)
+    }
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+enum IterState {
+    ARC,
+    LINES,
+    DONE,
+}
```

Comment:
Enum variants need to be camel case:
```suggestion
enum IterState {
    Arc,
    Lines,
    Done,
}
```